nemo_aligner/algorithms/dpo.py

```diff
@@ -196,10 +199,12 @@ def fit(self):
         self.run_timer.start_time()
 
         for _ in epoch_iter:
-            loop_iter = range(self.step, self.max_steps)
+            num_steps_in_epoch = min(
+                self.max_steps - self.step, self.num_steps_per_epoch - self.step % self.num_steps_per_epoch
+            )
+            loop_iter = range(num_steps_in_epoch)
```

Comment:
It's probably fine (so no need to revert it), but FYI I don't think you needed this change in dpo / supervised, because the loop below is over `zip(loop_iter, global_pbar)` which will stop when we reach the end of the train dataloader. This fix was specific to PPO due to the fact there was no such `zip`.

Reply:
Correct, and I did know this when I made the change, however, the old loop_iter was incorrect, as the train dataloader would be shorter in length and thus supercede the loop iter, but we should make the loop iter correct.

Although, thinking about it now, if max_steps is user set to be shorter than a single epoch, then wouldn't the loop_iter end up being shorter length than the train dataloader, in which case my change to loop iter is actually necessary?

Reply:
> the old loop_iter was incorrect, as the train dataloader would be shorter in length and thus supercede the loop iter, but we should make the loop iter correct.

I don't think it was "incorrect", since the code worked as intended (if the train dataloader is shorter, then we should stop earlier to end the epoch). It was potentially confusing though (took me a little while to figure out how this worked).

> f max_steps is user set to be shorter than a single epoch, then wouldn't the loop_iter end up being shorter length than the train dataloader, in which case my change to loop iter is actually necessary?

I do believe your change is indeed not necessary, but it should also be harmless (and it has the advantage of being more explicit). So I'm fine whether you keep it or revert it.